pkg/recipes/terraform/config/config.go

```diff
@@ -124,6 +153,72 @@ func (cfg *TerraformConfig) AddProviders(ctx context.Context, requiredProviders
 		cfg.Provider = providerConfigs
 	}
 
+	// Update module configuration with aliased provider names, if they exist.
+	logger.Info("Updating module providers with aliases")
+	if err := cfg.updateModuleWithProviderAliases(requiredProviders); err != nil {
+		return err
+	}
+
+	// Set the required providers for the Terraform configuration.
+	logger.Info("Update Terraform configuration with required providers")
+	if cfg.Terraform == nil {
+		cfg.Terraform = &TerraformDefinition{}
+	}
+	cfg.Terraform.RequiredProviders = requiredProviders
+
+	return nil
+}
+
+// updateModuleWithProviderAliases updates the module provider configuration in the Terraform config
+// by adding aliases to the provider configurations.
+func (cfg *TerraformConfig) updateModuleWithProviderAliases(requiredProviders map[string]*RequiredProviderInfo) error {
+	if cfg == nil {
+		return fmt.Errorf("terraform configuration is not initialized")
+	}
+	providerConfigs := cfg.Provider
+	moduleAliasConfig := map[string]string{}
+
+	for providerName, providerConfigList := range providerConfigs {
+		providerConfigDetails, ok := providerConfigList.([]map[string]any)
+		if !ok {
+			continue
+		}
+
+		// For each provider in the providerConfigs, if provider has a property "alias",
+		// add entry to the module provider configuration, per documentation:
+		// Provider configurations (those with the alias argument set) are never inherited automatically by modules,
+		// and so must always be passed explicitly using the providers map.
+		// https://developer.hashicorp.com/terraform/language/modules/develop/providers#legacy-shared-modules-with-provider-configurations
+
+		// Note: We're building configuration from user input, we're currently mapping the provider.alias names in
```

Comment:
> we're currently mapping the provider.alias names in

To make sure I understand - what do we mean by **currently** mapping? Is there a plan to follow up with another approach?